### PR TITLE
Fixes #179: Set up the Auth and Storage providers for Lumen

### LINKED ISSUE #179: CacheManager dependency resolving error
On the dev branch I am getting the following error:
`Unresolvable dependency resolving [Parameter #0 [ <required> $app ]] in class Illuminate\Cache\CacheManager`

```
in Container.php line 823
at Container->resolveNonClass(object(ReflectionParameter)) in Container.php line 798
at Container->getDependencies(array(object(ReflectionParameter)), array()) in Container.php line 771
at Container->build('Illuminate\Cache\CacheManager', array()) in Container.php line 626
at Container->make('Illuminate\Cache\CacheManager', array()) in Application.php line 393
at Application->make('Illuminate\Cache\CacheManager') in Container.php line 837
at Container->resolveClass(object(ReflectionParameter)) in Container.php line 800
at Container->getDependencies(array(object(ReflectionParameter)), array()) in Container.php line 771
at Container->build('Tymon\JWTAuth\Providers\Storage\Illuminate', array()) in Container.php line 626
at Container->make('Tymon\JWTAuth\Providers\Storage\Illuminate', array()) in Application.php line 393
at Application->make('Tymon\JWTAuth\Providers\Storage\Illuminate') in LumenServiceProvider.php line 207
at LumenServiceProvider->getConfigInstance('providers.storage') in LumenServiceProvider.php line 88
at LumenServiceProvider->Tymon\JWTAuth\Providers\{closure}(object(Application), array()) in Container.php line 733
at Container->build(object(Closure), array()) in Container.php line 626
at Container->make('tymon.jwt.provider.storage', array()) in Application.php line 393
at Application->make('tymon.jwt.provider.storage') in Container.php line 1157
at Container->offsetGet('tymon.jwt.provider.storage') in LumenServiceProvider.php line 139
at LumenServiceProvider->Tymon\JWTAuth\Providers\{closure}(object(Application), array()) in Container.php line 733
at Container->build(object(Closure), array()) in Container.php line 626
at Container->make('tymon.jwt.blacklist', array()) in Application.php line 393
at Application->make('tymon.jwt.blacklist') in Container.php line 1157
at Container->offsetGet('tymon.jwt.blacklist') in LumenServiceProvider.php line 101
at LumenServiceProvider->Tymon\JWTAuth\Providers\{closure}(object(Application), array()) in Container.php line 733
at Container->build(object(Closure), array()) in Container.php line 626
at Container->make('tymon.jwt.manager', array()) in Application.php line 393
at Application->make('tymon.jwt.manager') in Container.php line 1157
at Container->offsetGet('tymon.jwt.manager') in LumenServiceProvider.php line 126
at LumenServiceProvider->Tymon\JWTAuth\Providers\{closure}(object(Application), array()) in Container.php line 733
at Container->build(object(Closure), array()) in Container.php line 626
at Container->make('tymon.jwt.auth', array()) in Application.php line 393
at Application->make('tymon.jwt.auth', array()) in helpers.php line 38
at app('tymon.jwt.auth') in AuthController.php line 49
at AuthController->postLogin(object(Request))
at call_user_func_array(array(object(AuthController), 'postLogin'), array(object(Request))) in Container.php line 502
at Container->call(array(object(AuthController), 'postLogin'), array()) in Application.php line 1328
at Application->callControllerCallable(array(object(AuthController), 'postLogin'), array()) in Application.php line 1293
at Application->callLumenController(object(AuthController), 'postLogin', array(true, array('uses' => 'App\Http\Controllers\Auth\AuthController@postLogin'), array())) in Application.php line 1264
at Application->callControllerAction(array(true, array('uses' => 'App\Http\Controllers\Auth\AuthController@postLogin'), array())) in Application.php line 1232
at Application->callActionOnArrayBasedRoute(array(true, array('uses' => 'App\Http\Controllers\Auth\AuthController@postLogin'), array())) in Application.php line 1217
at Application->handleFoundRoute(array(true, array('uses' => 'App\Http\Controllers\Auth\AuthController@postLogin'), array())) in Application.php line 1138
at Application->Laravel\Lumen\{closure}() in Application.php line 1370
at Application->sendThroughPipeline(array(), object(Closure)) in Application.php line 1144
at Application->dispatch(null) in Application.php line 1084
at Application->run() in index.php line 28
```

### PR DESCRIPTION
**Pull request by ActuallyACat**
This seems to fix the Lumen issues with setting up the Storage and Auth providers not being resolved.

### CODE REVIEW

**Comment by ActuallyACat** [comment]
Fixes this issue: https://github.com/tymondesigns/jwt-auth/issues/179

**Comment by tymondesigns** [comment]
Although this would fix the default Lumen install, it would disable the ability to pass in other implementations.

I will fix this soon though

Thanks!

**Comment by jayhealey** [comment]
I encountered this issue after a `composer update` as well. 

It'd be great to see this fix merged in.

**Comment by ActuallyACat** [comment]
@jayhealey I've set the following up in my current project as a workaround: 
https://gist.github.com/ActuallyACat/a77143107e824bc786f2

Make sure you register the provider. Let me know if there's any problems! 

@tymondesigns is right, it would lock you into using AuthManager and CacheManager if it were to be merged in, so it's not the best solution - but it can be fine as a (temporary) bandaid solution :)

File: src/Providers/LumenServiceProvider.php [hunk 1]
```diff
@@ -83,9 +83,14 @@ protected function registerJWTProvider()
...
-        $this->app->singleton('tymon.jwt.provider.auth', function ($app) {
-            return $this->getConfigInstance('providers.auth');
-        });
+        $this->app->singleton(
+            'tymon.jwt.provider.auth',
+            function ($app) {
+                $provider = $this->config('providers.auth');
+
+                return $app->make($provider, [new AuthManager($app)]);
+            }
```
Comment (crynobone) [inline src/Providers/LumenServiceProvider.php]:
```php
return $app->make($provider, [$app->make('auth')]);
```

File: src/Providers/LumenServiceProvider.php [hunk 2]
```diff
@@ -83,9 +83,14 @@ protected function registerJWTProvider()
...
     {
-        $this->app->singleton('tymon.jwt.provider.auth', function ($app) {
-            return $this->getConfigInstance('providers.auth');
-        });
+        $this->app->singleton(
+            'tymon.jwt.provider.auth',
+            function ($app) {
+                $provider = $this->config('providers.auth');
+
+                return $app->make($provider, [new AuthManager($app)]);
```
Comment (crynobone) [inline src/Providers/LumenServiceProvider.php]:
```php
return $app->make($provider, [$app->make('auth')]);
```

File: src/Providers/LumenServiceProvider.php [hunk 3]
```diff
@@ -93,9 +98,14 @@ protected function registerAuthProvider()
...
     {
-        $this->app->singleton('tymon.jwt.provider.storage', function ($app) {
-            return $this->getConfigInstance('providers.storage');
-        });
+        $this->app->singleton(
+            'tymon.jwt.provider.storage',
+            function ($app) {
+                $provider = $this->config('providers.storage');
+
+                return $app->make($provider, [new CacheManager($app)]);
```
Comment (crynobone) [inline src/Providers/LumenServiceProvider.php]:
```php
return $app->make($provider, [$app->make('cache')]);
```

Comment (crynobone) [inline src/Providers/LumenServiceProvider.php]: This avoid generating a new instance instead of resolving with what already bind to the IoC.

**Pull request closed without merging**